include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
I think that having an array and false as return values would be bad for the usage. This, I think, is looking readable:

```
if (sanitized_gcontact($gcontact)) {
  $gcid = update_gcontact($gcontact);
  link_gcontact($gcid, $uid, $cid, $zcid);
}
```
Otherwise it would look like this:
```
$gcontact = sanitize_gcontact($gcontact);
if (is_array($gcontact)) {
  $gcid = update_gcontact($gcontact);
  link_gcontact($gcid, $uid, $cid, $zcid);
}
```
I don't like this ;-)